cms/djangoapps/contentstore/views/course.py

```diff
@@ -642,6 +642,8 @@ def _create_or_rerun_course(request):
             fields['display_name'] = display_name
 
         if 'source_course_key' in request.json:
+            fields['advertised_start'] = None
```

Comment:
I agree about your three possible places where this clearing can go:
- split.py - too low-level. Split performs general manipulation of fields and specific manipulation of children field. But no other specific fields are changed and we should keep it that way.
- task - This is potentially a better place than the view to clear the field. However, no other fields are set/cleared here yet. And perhaps the task should be lean and just perform the rerun without containing the logic of a restart.
- view - Feels about right, though a bit high-level. I'd prefer you moved the field clearing down into `_rerun_course` itself in case some other future view also reruns a course. That location puts the `advertised_date` clearing closer to the rerun task startup.